ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -806,6 +811,103 @@ func (ipfs *Connector) RepoStat(ctx context.Context) (*api.IPFSRepoStat, error)
 	return &stats, nil
 }
 
+// RepoGC performs a garbage collection sweep on the cluster peer's IPFS repo.
+func (ipfs *Connector) RepoGC(ctx context.Context) (*api.RepoGC, error) {
+	ctx, span := trace.StartSpan(ctx, "ipfsconn/ipfshttp/RepoGC")
+	defer span.End()
+
+	ctx1, cancel1 := context.WithTimeout(ctx, ipfs.config.IPFSRequestTimeout)
+	defer cancel1()
+
+	id := &api.ID{}
+	err := ipfs.rpcClient.CallContext(
+		ctx1,
+		"",
+		"Cluster",
+		"ID",
+		struct{}{},
+		id,
+	)
+
+	repoGC := api.RepoGC{
+		Peer:     id.ID,
+		Peername: id.Peername,
+		Keys:     make([]api.IPFSRepoGC, 0),
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	outGCs := make(chan string)
+	go func() {
+		lastRefTime := time.Now()
+		ticker := time.NewTicker(progressTick)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				if time.Since(lastRefTime) >= ipfs.config.IPFSRequestTimeout {
+					cancel() // timeout
+					return
+				}
+			case <-outGCs:
+				lastRefTime = time.Now()
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	keys, err := ipfs.repoGCProgress(ctx, outGCs)
+	repoGC.Keys = keys
+
+	return &repoGC, err
+}
+
+func (ipfs *Connector) repoGCProgress(ctx context.Context, out chan<- string) ([]api.IPFSRepoGC, error) {
```

Comment:
Why did you add this? Our previous convo about exactly this here:

![image](https://user-images.githubusercontent.com/1027022/60991048-0c40ee00-a34a-11e9-8587-ac6bacea3ada.png)

In the last review I did not raise any comments about this part needing a full rework. Adopting this approach here does not help because the GC output is not likely to hang long once you start getting it. Most of the wait will happen with no output at all. This overcomplicates. Revert to the previous form please.

Please next time, concentrate on resolving review comments. Do not move forward into reworking full parts without asking first or being asked to.

Reply:
Have removed checking timeout with progress (`repoGCProgress`)